src/mbgl/map/vector_tile_data.cpp

```diff
@@ -144,15 +144,15 @@ void VectorTileData::redoPlacement(const PlacementConfig newConfig, const std::f
     if (newConfig != placedConfig) {
         targetConfig = newConfig;
 
-        if (!workRequest) {
-            // Don't start a new placement request when the current one hasn't completed yet, or when
-            // we are parsing buckets.
-            redoPlacement(callback);
-        }
+        redoPlacement(callback);
     }
 }
 
 void VectorTileData::redoPlacement(const std::function<void()>& callback) {
+    // Don't start a new placement request when the current one hasn't completed yet, or when
+    // we are parsing buckets.
+    if (workRequest) return;
+
     workRequest.reset();
```

Comment:
It's safe to remove this one; `if (workRequest) return;` guarantees that `reset()` would be a no-op. For the other ones, there's a difference:

* reset, create new: Ensures the current task is ended and the callback for it will not be called, then starts a new one.
* create new, assign: The callback for the previous task could still be executed, which might put things into an unexpected state.